src/tfvc/tfvc.ts

```diff
@@ -139,24 +137,84 @@ export class Tfvc {
             TfvcOutput.AppendLine(`tf ${args.GetArgumentsForDisplay()}`);
         }
 
-        return cp.spawn(this._tfvcPath, args.GetArguments(), options);
+        return await this._tfRunner.Run(args, options, this._isExe);
     }
+}
 
-    private async _exec(args: IArgumentProvider, options: any = {}): Promise<IExecutionResult> {
-        const child = this.spawn(args, options);
+class TfRunner implements IDisposable {
+    private _location: string;
+    private _options: any;
+    private _runningInstance: cp.ChildProcess;
 
-        if (options.input) {
-            child.stdin.end(options.input, "utf8");
+    public constructor(location: string, options: any) {
+        this._location = location;
+        this._options = options;
+    }
+
+    public dispose() {
+        if (this._runningInstance) {
+            this._runningInstance.kill();
+            this._runningInstance = undefined;
         }
+    }
+
+    public async Run(args: IArgumentProvider, options: any, isExe: boolean): Promise<IExecutionResult> {
+        const start: number = new Date().getTime();
+        const tfInstance: cp.ChildProcess = await this.getMatchingTfInstance(options);
+        // now that we have the matching one, start a new process (but don't wait on it to finish)
+        this.startNewTfInstance(options);
+
+        // Use the tf instance to perform the command
+        const argsForStandardInput: string = args.GetCommandLine();
+        const result: IExecutionResult = await this.runCommand(argsForStandardInput, tfInstance, isExe);
+
+        // log the results
+        const end: number = new Date().getTime();
+        Logger.LogDebug(`TFVC: ${args.GetCommand()} exit code: ${result.exitCode} (duration: ${end - start}ms)`);
 
-        const result = await Tfvc.execProcess(child);
-        Logger.LogDebug(`TFVC exit code: ${result.exitCode}`);
         return result;
     }
 
-    private static async execProcess(child: cp.ChildProcess): Promise<IExecutionResult> {
+    private async getMatchingTfInstance(options: any): Promise<cp.ChildProcess> {
+        if (!this._runningInstance || !this.optionsMatch(options, this._options)) {
+            if (this._runningInstance) {
+                this._runningInstance.kill();
+            }
+            // spawn a new instance of TF with these options
+            await this.startNewTfInstance(options);
+        }
+        return this._runningInstance;
+    }
+
+    private async startNewTfInstance(options: any) {
+        // Start up a new instance of TF for later use
+        this._options = options;
+        this._runningInstance = await this.spawn(options);
+    }
+
+    private optionsMatch(options1: any, options2: any): boolean {
+        return (!options1 && !options2) || (options1.cwd === options2.cwd);
```

Comment:
enough? for now yes, the only thing that makes the command line launched invalid for the current workspace is if the working directory changed for some reason. If a new folder is opened, the tfrunner is disposed and a new one created, so that should work fine.